File: carbon/src/main/java/carbon/view/TextAppearanceView.java
package carbon.view;

import android.content.res.ColorStateList;
import android.graphics.Typeface;
import android.text.TextPaint;

import androidx.annotation.NonNull;

public interface TextAppearanceView {
    void setTypeface(Typeface typeface, int style);

    void setAllCaps(boolean allCaps);

    void setTextColor(@NonNull ColorStateList textColor);

    void setTextSize(float size);

    CharSequence getText();

    TextPaint getPaint();

    void setText(CharSequence text);
}

File: carbon/src/main/java/carbon/view/TextAppearanceView.kt
package carbon.view

import android.content.res.ColorStateList
import android.graphics.Typeface
import android.text.TextPaint

interface TextAppearanceView {

    var text: CharSequence

    val paint: TextPaint

    fun setTypeface(typeface: Typeface, style: Int)

    fun setAllCaps(allCaps: Boolean)

    fun setTextColor(textColor: ColorStateList)

    fun setTextSize(size: Float)
}
